### PR TITLE
rewrite mozo.ui from scratch

### PR DESCRIPTION
**Pull request by mbkma**
fixing all but one deprecation. The look should be almost identical. The only real change I made is that you can not slide the buttons on the right out of the window.
I wrote this from scratch so I hope I re-added all signals in Glade, would be nice if you could double-check (i.e. check if all buttons, drag and drop, ... work).

### CODE REVIEW

File: data/mozo.ui [hunk 1]
```diff
@@ -1,7 +1,167 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!-- Generated with glade 3.22.1 -->
+<!-- Generated with glade 3.38.2 -->
 <interface>
-  <requires lib="gtk+" version="3.14"/>
+  <requires lib="gtk+" version="3.24"/>
+  <object class="GtkImage" id="cancel-icon">
+    <property name="visible">True</property>
+    <property name="can-focus">False</property>
+    <property name="icon-name">gtk-cancel</property>
```
Comment (rbuj) [inline data/mozo.ui:8]: We use process-stop as replacement for gtk-cancel.

File: data/mozo.ui [hunk 2]
```diff
@@ -1,7 +1,167 @@
...
+  </object>
+  <object class="GtkImage" id="close-icon">
+    <property name="visible">True</property>
+    <property name="can-focus">False</property>
+    <property name="icon-name">gtk-close</property>
```
Comment (rbuj) [inline data/mozo.ui:13]: We use window-close as replacement for gtk-close.
https://developer.gnome.org/gtk3/stable/gtk3-Stock-Items.html#GTK-STOCK-CLOSE:CAPS

File: data/mozo.ui [hunk 3]
```diff
@@ -1,7 +1,167 @@
...
+  </object>
+  <object class="GtkImage" id="delete-icon">
+    <property name="visible">True</property>
+    <property name="can-focus">False</property>
+    <property name="icon-name">gtk-delete</property>
```
Comment (rbuj) [inline data/mozo.ui:18]: We use edit-delete as replacement for gtk-delete.
https://developer.gnome.org/gtk3/stable/gtk3-Stock-Items.html#GTK-STOCK-DELETE:CAPS

File: data/mozo.ui [hunk 4]
```diff
@@ -1,7 +1,167 @@
...
+  </object>
+  <object class="GtkImage" id="help-icon">
+    <property name="visible">True</property>
+    <property name="can-focus">False</property>
+    <property name="icon-name">help</property>
```
Comment (rbuj) [inline data/mozo.ui:23]: We use help-browser as replacement for gtk-help.
https://developer.gnome.org/gtk3/stable/gtk3-Stock-Items.html#GTK-STOCK-HELP:CAPS

File: data/mozo.ui [hunk 5]
```diff
@@ -1,7 +1,167 @@
...
+  </object>
+  <object class="GtkImage" id="move-up-icon">
+    <property name="visible">True</property>
+    <property name="can-focus">False</property>
+    <property name="icon-name">up</property>
+  </object>
+  <object class="GtkImage" id="new-item-icon">
+    <property name="visible">True</property>
+    <property name="can-focus">False</property>
+    <property name="icon-name">gtk-add</property>
```
Comment (rbuj) [inline data/mozo.ui:38]: We use list-add as replacement for gtk-add.
https://developer.gnome.org/gtk3/stable/gtk3-Stock-Items.html#GTK-STOCK-ADD:CAPS

File: data/mozo.ui [hunk 6]
```diff
@@ -1,7 +1,167 @@
...
+  </object>
+  <object class="GtkImage" id="new-menu-icon">
+    <property name="visible">True</property>
+    <property name="can-focus">False</property>
+    <property name="icon-name">gtk-new</property>
```
Comment (rbuj) [inline data/mozo.ui:43]: We use document-new as replacement for gtk-new.
https://developer.gnome.org/gtk3/stable/gtk3-Stock-Items.html#GTK-STOCK-NEW:CAPS

File: data/mozo.ui [hunk 7]
```diff
@@ -1,7 +1,167 @@
...
+  </object>
+  <object class="GtkImage" id="properties-icon">
+    <property name="visible">True</property>
+    <property name="can-focus">False</property>
+    <property name="icon-name">document-properties</property>
+  </object>
+  <object class="GtkImage" id="redo-icon">
+    <property name="visible">True</property>
+    <property name="can-focus">False</property>
+    <property name="icon-name">redo</property>
```
Comment (rbuj) [inline data/mozo.ui:53]: we use edit-redo
https://developer.gnome.org/gtk3/stable/gtk3-Stock-Items.html#GTK-STOCK-REDO:CAPS

File: data/mozo.ui [hunk 8]
```diff
@@ -1,7 +1,167 @@
...
+  </object>
+  <object class="GtkImage" id="revert-icon">
+    <property name="visible">True</property>
+    <property name="can-focus">False</property>
+    <property name="icon-name">revert</property>
```
Comment (rbuj) [inline data/mozo.ui:58]: we use document-revert
https://developer.gnome.org/gtk3/stable/gtk3-Stock-Items.html#GTK-STOCK-REVERT-TO-SAVED:CAPS

File: data/mozo.ui [hunk 9]
```diff
@@ -37,701 +197,416 @@
       </popup>
     </ui>
   </object>
-  <object class="GtkMenu" constructor="uimanager1" id="edit_menu">
+  <object class="GtkImage" id="undo-icon">
     <property name="visible">True</property>
-    <property name="can_focus">False</property>
+    <property name="can-focus">False</property>
+    <property name="icon-name">undo</property>
```
Comment (rbuj) [inline data/mozo.ui:203]: we use edit-undo
https://developer.gnome.org/gtk3/stable/gtk3-Stock-Items.html#GTK-STOCK-UNDO:CAPS

File: data/mozo.ui [hunk 10]
```diff
@@ -1,7 +1,167 @@
...
+        </child>
+        <child>
+          <object class="GtkBox">
+            <property name="visible">True</property>
+            <property name="can-focus">False</property>
+            <child>
+              <object class="GtkImage">
+                <property name="visible">True</property>
+                <property name="can-focus">False</property>
+                <property name="icon-name">gtk-dialog-question</property>
```
Comment (rbuj) [inline data/mozo.ui:128]: we use dialog-question
https://developer.gnome.org/gtk3/stable/gtk3-Stock-Items.html#GTK-STOCK-DIALOG-QUESTION:CAPS

**Review by rbuj — CHANGES_REQUESTED** [review]
It upgrades the properties names. It fixes the deprecation of stock buttons. It remove unused ids from mozo.ui. It refactors two functions names.

It unnecessarily maps 3 new response ids returned by the dialog, having a handler.

**Review by rbuj — APPROVED** [review]
LGTM

**Comment by lukefromdc** [comment]
I just build and installed this, though I did not see this in time to test it pre-merge. Works fine so no problems there. At first I got a fistfull of runtime warnings, but those turned out to come from an invalid entry at the top of the menu. Once that was removed, Mozo worked perfectly.

**Comment by raveit65** [comment]
@mbkma 
I got a report about gtk3 version in mozo.ui with Rocky8 distro.
https://bugzilla.redhat.com/show_bug.cgi?id=2136074
For what reason gtk3 version was bumped to 3.24?
Do you think we can lower version to 3.22 ?

Beside from this i got a warning in glade.

Bildschirmfoto zu 2022-10-23 21-20-47

**Comment by raveit65** [comment]
@mbkma 
You are the author of this commit. Is it possible to lower gtk+ required version of mozo.ui ?

**Comment by mbkma** [comment]
yeah, this were just the standard settings in glade. See https://github.com/mate-desktop/mozo/pull/88.

**Pull request merged**